pkg/iobuf/bufferedreaderseeker.go

```diff
@@ -82,9 +82,6 @@ func NewBufferedReaderSeeker(r io.Reader) *BufferedReadSeeker {
 	)
 
 	seeker = asSeeker(r)
-	if seeker == nil {
```

Comment:
This was removed to avoid checking out a buffer eagerly. We already [lazily](https://github.com/trufflesecurity/trufflehog/blob/cf54b71a9410df1960bad536b227958e776ace71/pkg/iobuf/bufferedreaderseeker.go#L123) check out a buffer when needed.